crypto/crypto/src/ed25519.rs

```diff
@@ -214,6 +218,19 @@ impl SigningKey for Ed25519PrivateKey {
     type VerifyingKeyMaterial = Ed25519PublicKey;
     type SignatureMaterial = Ed25519Signature;
 
+    fn sign<T: CryptoHash + Serialize>(
+        &self,
+        message: &T,
+    ) -> Result<Ed25519Signature, CryptoMaterialError> {
+        let mut bytes = <T::Hasher as CryptoHasher>::seed().to_vec();
+        lcs::serialize_into(&mut bytes, &message)
+            .map_err(|_| CryptoMaterialError::SerializationError)?;
```

Comment:
FYI in the rest of the codebase, we assume that LCS serialization into a vector never fails. (From the back of my mind, the only way it would fail is because of a non-supported type such as `char`, `f32`, `f64` or a clowny implementation of a `Serialize` trait for some user type.)

Reply:
I'll let you and @kchalkias figure it out in a forthcoming PR, but I think that an arcane failure of `Serialization`, if it manifests as a signature failure, is sufficiently far from its root cause to warrant reporting on the error precisely, here.